workflows/wf_theiacov_illumina_pe.wdl

```diff
@@ -122,18 +128,34 @@ workflow theiacov_illumina_pe {
   # run organism-specific typing
   if (organism == "MPXV" || organism == "sars-cov-2" || organism == "flu" && select_first([abricate_flu.run_nextclade]) ) { 
     # tasks specific to either MPXV, sars-cov-2, or flu
-    call taxon_ID.nextclade_one_sample {
+    call taxon_ID.nextclade_one_sample as nextclade_one_sample_run1 {
       input:
-        genome_fasta = select_first([consensus.consensus_seq, irma.seg4_ha_assembly]),
-        dataset_name = select_first([abricate_flu.nextclade_name, nextclade_dataset_name, organism]),
-        dataset_reference = select_first([abricate_flu.nextclade_ref, nextclade_dataset_reference]),
-        dataset_tag = select_first([abricate_flu.nextclade_ds_tag, nextclade_dataset_tag])
+        genome_fasta = select_first([consensus.consensus_seq, irma.seg_ha_assembly]),
+        dataset_name = select_first([abricate_flu.nextclade_name_ha, nextclade_dataset_name, organism]),
+        dataset_reference = select_first([abricate_flu.nextclade_ref_ha, nextclade_dataset_reference]),
+        dataset_tag = select_first([abricate_flu.nextclade_ds_tag_ha, nextclade_dataset_tag])
     }
-    call taxon_ID.nextclade_output_parser_one_sample {
+    call taxon_ID.nextclade_output_parser_one_sample as nextclade_output_parser_one_sample_run1 {
       input:
-      nextclade_tsv = nextclade_one_sample.nextclade_tsv,
+      nextclade_tsv = nextclade_one_sample_run1.nextclade_tsv,
       organism = organism
     }
+  }
+    if (organism == "flu" && select_first([abricate_flu.run_nextclade]) ) { 
+    # tasks specific to flu NA - run nextclade a second time
+    call taxon_ID.nextclade_one_sample as nextclade_one_sample_run2{
```

Comment:
Oh, nice! I like this solution of having a `run2` task. 

Wondering if the second conditional should be based on the `abricate_flu.run_nextclade` variable or if this second run could be triggered by the existence of the optional `seg_na_assembly`. With this, you may also be able to set the genome_fasta as the NA segment--may still need the select_first function because of the whole optional-input chaos of WDL, e.g. 

```
if (organism == "flu" && defined(irma.seg_na_assembly)) ) {
    # tasks specific to flu NA - run nextclade a second time
    call taxon_ID.nextclade_one_sample as nextclade_one_sample_run2{
    input:
        genome_fasta = select_first([irma.seg_na_assembly]),
        dataset_name = select_first([abricate_flu.nextclade_name_na, nextclade_dataset_name, organism]),
...
}
```